src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/EditorConfigNamingStyleParser.cs

```diff
@@ -0,0 +1,251 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using static Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles.SymbolSpecification;
+
+namespace Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles
+{
+    internal static partial class EditorConfigNamingStyleParser
+    {
+        private static Dictionary<int, string> namingStyleStringCache = new Dictionary<int, string>();
+
+        public static string GetNamingStylesStringFromDictionary(IReadOnlyDictionary<string, object> allRawConventions)
+        {
+            var hashcode = allRawConventions.GetHashCode();
```

Comment:
Who is to say that the hash is usable this way? You're assuming that both [a] the hash code is implemented in a meaningful way and [b] that hashcodes being equal implies equality.